core/domain/stats_jobs_one_off.py

```diff
@@ -914,3 +914,151 @@ def reduce(key, stringified_values):
                     'all: %s sum:%s' % (
                         key, state_name, all_state_hit[state_name],
                         sum_state_hit[state_name]),)
+
+
+class GenerateMissingStatsModelsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Generates state stats models for explorations which do not have events
+    recorded. This should be run after GenerateV1StatisticsJob has completed a
+    successful run.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(exp_model):
+        """Implements the map function. Must be declared @staticmethod.
+
+        Args:
+            exp_model: ExplorationModel.
+        """
+        if exp_model.deleted:
+            return
+
+        try:
+            exploration = exp_services.get_exploration_from_model(exp_model)
+        except Exception as e:
+            # Exploration does not exist.
+            return
+
+        latest_exp_version = exploration.version
+        exp_id = exploration.id
+
+        version_numbers = range(1, latest_exp_version + 1)
+        try:
+            explorations_by_version = (
+                exp_services.get_multiple_explorations_by_version(
+                    exp_id, version_numbers))
+        except Exception as e:
+            yield (exp_id, str(e))
+            return
+
+        exp_stats_by_version = (
+            stats_services.get_multiple_exploration_stats_by_version(
+                exp_id, version_numbers))
+        versions_with_no_stats = []
+        versions_with_stats = []
+        for index, per_version_stats in enumerate(exp_stats_by_version):
+            if per_version_stats is None:
+                versions_with_no_stats.append(index + 1)
+            else:
+                versions_with_stats.append(index + 1)
+        if versions_with_no_stats and versions_with_stats:
+            yield (
+                exp_id,
+                'ERROR: Stats models incompletely generated, has stats %s, '
+                'has no stats %s' % (
+                    versions_with_stats, versions_with_no_stats))
+            return
+
+        if versions_with_stats:
```

Comment:
Doesn't this have to be an 'elif'?

Reply:
No, because of the "return" in the previous clause?

Reply:
Ah. My bad. Seems good to go!